From 4e0b8bd13bbfc5d52b87a673fcf563cbd715bb17 Mon Sep 17 00:00:00 2001
From: Patrik2 Carlsson <patrik2.carlsson@sonymobile.com>
Date: Tue, 27 Jan 2015 16:49:45 +0100
Subject: [PATCH 21/95] Play timed text even after 35.79 minutes

int overflow when converting current position in ms to Us

Change-Id: I12882e52ed1ac92644d601efae5c10ec25ae9e63
---
 media/libmediaplayerservice/nuplayer/NuPlayer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index aeea204..252b643 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -1841,7 +1841,7 @@ void NuPlayer::onSourceNotify(const sp<AMessage> &msg) {
             int posMs;
             int64_t timeUs, posUs;
             driver->getCurrentPosition(&posMs);
-            posUs = posMs * 1000;
+            posUs = (int64_t) posMs * 1000ll;
             CHECK(buffer->meta()->findInt64("timeUs", &timeUs));
 
             if (posUs < timeUs) {
-- 
2.5.0

